integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiUpdateDomainConfig.java

```diff
@@ -248,15 +249,39 @@ public void beforeEach() {
     }
   }
 
+  /**
+   * Check the environment variable with special characters.
+   */
+  @Test
+  @Order(0)
+  @DisplayName("Check environment variable with special characters")
+  public void testMiiSpecialEnv() {
+    Domain domain1 = assertDoesNotThrow(() -> getDomainCustomResource(domainUid, domainNamespace),
+        String.format("getDomainCustomResource failed with ApiException when tried to get domain %s in namespace %s",
+            domainUid, domainNamespace));
+    List<V1EnvVar> envList = domain1.getSpec().getServerPod().getEnv();
+
+    boolean found = false;
+    for (int i = 0; i < envList.size(); i++) {
+      logger.info("The name is: {0}, value is: {1}", envList.get(i).getName(), envList.get(i).getValue());
+      if (envList.get(i).getName().equalsIgnoreCase("SPECIAL_ENV")) {
+        assertTrue(
+            envList.get(i).getValue().equalsIgnoreCase("${DOMAIN_UID}~##!'%*$(ls)"),
+            "Expected value for SPECIAL_ENV variable does not mtach");
+        found = true;
+      }
```

Comment:
To complete the testing of env we need to see if WL configuration that references the env at runtime ends up with expected value:
- Reference the env in a macro within WL configuration (such as in the notes field of some bean), and verify using WLST or JMX that the deployed domain's bean attribute contains the expected value.
- Do this for both 'model YAML' and 'sit-cfg'.  (They have a different macro format, but each can reference an env var.)
(See the JIRA for the combinatorial matrix of test use cases.)

Reply:
Added configuration check for custom environment inside domain configuration. 
We will address situational config usecase in different PR